packages/test/local-server-stress-tests/src/localServerStressHarness.ts

```diff
@@ -0,0 +1,1268 @@
+			// as synchronization + eventual consistency validation should be idempotent.
+			// 0.5 is arbitrary but there's no reason anyone should want a probability near this.
+			assert(validationStrategy.probability < 0.5, "Use a lower synchronization probability.");
+			generatorFactory = (): AsyncGenerator<TOperation | Synchronize, TState> => {
+				const baseGenerator = model.generatorFactory();
+				return async (state: TState): Promise<TOperation | Synchronize | typeof done> =>
+					!state.isDetached && state.random.bool(validationStrategy.probability)
+						? { type: "synchronize" }
+						: baseGenerator(state);
+			};
+			break;
+		}
+
+		case "fixedInterval": {
+			generatorFactory = (): AsyncGenerator<TOperation | Synchronize, TState> => {
+				const baseGenerator = model.generatorFactory();
+				return interleaveAsync<TOperation | Synchronize, TState>(
+					baseGenerator,
+					async (state) =>
+						state.isDetached ? baseGenerator(state) : ({ type: "synchronize" } as const),
+					validationStrategy.interval,
+					1,
+					ExitBehavior.OnEitherExhausted,
+				);
+			};
+			break;
+		}
+
+		case "partialSynchronization": {
+			// passing 1 here causes infinite loops. passing close to 1 is wasteful
+			// as synchronization + eventual consistency validation should be idempotent.
+			// 0.5 is arbitrary but there's no reason anyone should want a probability near this.
+			assert(validationStrategy.probability < 0.5, "Use a lower synchronization probability.");
+			generatorFactory = (): AsyncGenerator<TOperation | Synchronize, TState> => {
+				const baseGenerator = model.generatorFactory();
+				return async (state: TState): Promise<TOperation | Synchronize | typeof done> => {
+					if (!state.isDetached && state.random.bool(validationStrategy.probability)) {
+						const selectedClients = new Set(
+							state.clients
+								.filter(
+									(client) => client.container.connectionState === ConnectionState.Connected,
+								)
+								.filter(() => state.random.bool(validationStrategy.clientProbability)),
+						);
+
+						return { type: "synchronize", clients: [...selectedClients] };
+					} else {
+						return baseGenerator(state);
+					}
+				};
+			};
+			break;
+		}
+		default: {
+			unreachableCase(validationStrategy);
+		}
+	}
+
+	const minimizationTransforms = model.minimizationTransforms as
+		| MinimizationTransform<TOperation | Synchronize>[]
+		| undefined;
+
+	const isSynchronizeOp = (op: BaseOperation): op is Synchronize => op.type === "synchronize";
+	const reducer: AsyncReducer<TOperation | Synchronize, TState> = async (state, operation) => {
+		// TODO: Only synchronize listed clients if specified
+		if (isSynchronizeOp(operation)) {
+			const connectedClients = state.clients.filter(
+				(client) => client.container.connectionState === ConnectionState.Connected,
+			);
+
+			await Promise.all(
+				connectedClients.map(
+					async (c) =>
+						new Promise<void>((resolve) =>
```

Comment:
seems like this check wouldn't be sufficient for consistency--we'd need to do another wait to ensure all containers have caught up to container with the latest seq after this promise resolves